src/rust/engine/hashing/src/lib.rs

```diff
@@ -128,6 +134,12 @@ impl Serialize for Digest {
   }
 }
 
+impl Default for Digest {
```

Comment:
This is ... vaguely dangerous maybe? Or not. I'm imagining a situation where we add `MyStruct::default()`, and empty with an invalid value. Maybe since digests are always either consumed or checked, you would "always" notice an incorrect digest? And EMPTY_DIGEST actually really means "the empty Snapshot", which is close to the meaning of `Default`?

Reply:
Removed them